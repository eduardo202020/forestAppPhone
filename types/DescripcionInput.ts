/* eslint-disable no-unused-vars */
export interface DescripcionInputProps {
  value: string;
  onChange: (value: string) => void;
  editable?: boolean;
  error?: string;
}
