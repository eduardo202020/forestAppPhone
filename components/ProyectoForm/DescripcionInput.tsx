import React from 'react';
import { TextInput, Text, StyleSheet } from 'react-native';
import { DescripcionInputProps } from '../../types/DescripcionInput';

const DescripcionInput: React.FC<DescripcionInputProps> = ({
  value,
  onChange,
  editable = true,
  error,
}) => (
  <>
    <TextInput
      style={[styles.input, styles.textArea]}
      placeholder="DescripciÃ³n"
      value={value}
      onChangeText={onChange}
      editable={editable}
      multiline
      numberOfLines={4}
    />
    {error && <Text style={styles.error}>{error}</Text>}
  </>
);

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 5,
    padding: 10,
    marginBottom: 10,
  },
  textArea: {
    height: 80,
    textAlignVertical: 'top',
  },
  error: {
    color: 'red',
    fontSize: 12,
  },
});

export default DescripcionInput;
