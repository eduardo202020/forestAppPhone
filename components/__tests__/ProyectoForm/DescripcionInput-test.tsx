import React from 'react';
import renderer from 'react-test-renderer';
import DescripcionInput from '../../ProyectoForm/DescripcionInput';

describe('DescripcionInput', () => {
  it('se renderiza sin errores', () => {
    const tree = renderer
      .create(
        <DescripcionInput
          value="desc"
          onChange={() => {}}
          editable
          error={undefined}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('muestra el error si existe', () => {
    const tree = renderer
      .create(
        <DescripcionInput
          value=""
          onChange={() => {}}
          editable
          error="Campo requerido"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('llama a onChange cuando el usuario escribe', () => {
    const onChange = jest.fn();
    const testRenderer = renderer.create(
      <DescripcionInput
        value=""
        onChange={onChange}
        editable
        error={undefined}
      />
    );
    const input = testRenderer.root.findByProps({ placeholder: 'Descripci√≥n' });
    input.props.onChangeText('nuevo texto');
    expect(onChange).toHaveBeenCalledWith('nuevo texto');
  });
});
