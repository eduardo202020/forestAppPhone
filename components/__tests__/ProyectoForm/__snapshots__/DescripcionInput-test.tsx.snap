// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescripcionInput muestra el error si existe 1`] = `
[
  <TextInput
    editable={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Descripción"
    style={
      [
        {
          "backgroundColor": "#f9f9f9",
          "borderColor": "#ccc",
          "borderRadius": 8,
          "borderWidth": 1,
          "fontSize": 16,
          "marginBottom": 12,
          "padding": 10,
        },
        {
          "minHeight": 60,
        },
      ]
    }
    value=""
  />,
  <Text
    style={
      {
        "color": "red",
        "fontSize": 13,
        "marginBottom": 8,
      }
    }
  >
    Campo requerido
  </Text>,
]
`;

exports[`DescripcionInput se renderiza sin errores 1`] = `
<TextInput
  editable={true}
  multiline={true}
  onChangeText={[Function]}
  placeholder="Descripción"
  style={
    [
      {
        "backgroundColor": "#f9f9f9",
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "fontSize": 16,
        "marginBottom": 12,
        "padding": 10,
      },
      {
        "minHeight": 60,
      },
    ]
  }
  value="desc"
/>
`;
